src/sentry/replays/consumers/recording/process_recording.py

```
@@ -0,0 +1,212 @@
+            key=cache_key,
+            id=recording_segment_uuid,
+            chunk_index=chunk_index,
+            chunk_data=message_dict["payload"],
+            timeout=CACHE_TIMEOUT,
+        )
+
+    def _process_headers(
+        self, recording_segment_with_headers: bytes
+    ) -> tuple[RecordingSegmentHeaders, bytes]:
+        # split the recording payload by a newline into the headers and the recording
+        try:
+            recording_headers, recording_segment = recording_segment_with_headers.split(b"\n", 1)
+        except ValueError:
+            raise MissingRecordingSegmentHeaders
+        return json.loads(recording_headers), recording_segment
+
+    def _store(
+        self,
+        message_dict: RecordingSegmentMessage,
+        cached_replay_recording_segment: CachedAttachment,
+    ) -> None:
+        try:
+            headers, recording_segment = self._process_headers(cached_replay_recording_segment.data)
+        except MissingRecordingSegmentHeaders:
+            logger.warning(f"missing header on {message_dict['replay_id']}")
+            return
+
+        # create a File for our recording segment.
+        recording_segment_file_name = f"rr:{message_dict['replay_id']}:{headers['sequence_id']}"
+        file = File.objects.create(
+            name=recording_segment_file_name,
+            type="replay.recording",
+        )
+        file.putfile(
+            BytesIO(recording_segment),
+            blob_size=settings.SENTRY_ATTACHMENT_BLOB_SIZE,
+        )
+        # associate this file with an indexable replay_id via ReplayRecordingSegment
+        ReplayRecordingSegment.objects.create(
+            replay_id=message_dict["replay_id"],
+            project_id=message_dict["project_id"],
+            sequence_id=headers["sequence_id"],
+            file_id=file.id,
+        )
+        # delete the recording segment from cache after we've stored it
+        cached_replay_recording_segment.delete()
```

We should suppress exceptions here since this operation is ostensibly completed.

---

we don't suppress exceptions elsewhere where we call delete -- can monitor and see if exceptions happen at all here.